src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,74 @@
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { useEffect, useState } from 'react';
+import { getPeople } from '../api';
+import { transformPeopleData } from '../utils/transformPeopleData';
+import { Person } from '../types';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [error, setError] = useState(false);
+
+  const loadPeople = async () => {
+    setIsLoading(true);
+    setError(false);
+
+    try {
+      const peopleFromServer = await getPeople();
+
+      const peopleWithParents = transformPeopleData(peopleFromServer);
+
+      setPeople(peopleWithParents);
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+      setIsLoaded(true);
+    }
+  };
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+
+      <div className="block">
+        <div className="columns is-desktop is-flex-direction-row-reverse">
+          <div className="column is-7-tablet is-narrow-desktop">
+            <PeopleFilters />
+          </div>
+
+          <div className="column">
+            <div className="box table-container">
+              {isLoading && <Loader />}
+
+              {error && (
+                <p data-cy="peopleLoadingError">Something went wrong</p>
+              )}
+
+              {!isLoading && isLoaded && people && people.length === 0 && (
+                <p data-cy="noPeopleMessage">
+                  There are no people on the server
+                </p>
+              )}
+
+              {!isLoading && isLoaded && people && people.length === 0 && (
+                <p>There are no people matching the current search criteria</p>
+              )}
```

Comment:
There is a redundancy in the conditions for displaying messages when there are no people. Both conditions check for '!isLoading && isLoaded && people && people.length === 0'. Consider consolidating these into a single condition to avoid duplication.